From 654fcc1546bc173732f8f18849fccd8fc65a2122 Mon Sep 17 00:00:00 2001
From: Dezhi Kong <dezhi.kong@amlogic.com>
Date: Tue, 11 Feb 2020 16:18:49 +0800
Subject: [PATCH 6/6] vpp: add endian process sync with Q [1/1]

PD#SWPL-20420

Problem:
add endian process sync with Q

Solution:
add endian process sync with Q

Verify:
on U212

Change-Id: I562c1532461688a1cda5bb1d42863730cfc155d0
Signed-off-by: Dezhi Kong <dezhi.kong@amlogic.com>
---
 drivers/amlogic/media/video_sink/video_hw.c | 30 +++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/drivers/amlogic/media/video_sink/video_hw.c b/drivers/amlogic/media/video_sink/video_hw.c
index 88fc6b358cd1..874ca6c26bad 100644
--- a/drivers/amlogic/media/video_sink/video_hw.c
+++ b/drivers/amlogic/media/video_sink/video_hw.c
@@ -740,6 +740,14 @@ static void vd1_set_dcu(
 		VSYNC_WR_MPEG_REG_BITS(
 			G12_VD1_IF0_GEN_REG3,
 			(burst_len & 0x3), 1, 2);
+		if (vf->flag & VFRAME_FLAG_VIDEO_LINEAR)
+			VSYNC_WR_MPEG_REG_BITS(
+				G12_VD1_IF0_GEN_REG3,
+				0, 0, 1);
+		else
+			VSYNC_WR_MPEG_REG_BITS(
+				G12_VD1_IF0_GEN_REG3,
+				1, 0, 1);
 		if (is_mvc) {
 			VSYNC_WR_MPEG_REG_BITS(
 				G12_VD2_IF0_GEN_REG3,
@@ -747,6 +755,14 @@ static void vd1_set_dcu(
 			VSYNC_WR_MPEG_REG_BITS(
 				G12_VD2_IF0_GEN_REG3,
 				(burst_len & 0x3), 1, 2);
+			if (vf->flag & VFRAME_FLAG_VIDEO_LINEAR)
+				VSYNC_WR_MPEG_REG_BITS(
+					G12_VD2_IF0_GEN_REG3,
+					0, 0, 1);
+			else
+				VSYNC_WR_MPEG_REG_BITS(
+					G12_VD2_IF0_GEN_REG3,
+					1, 0, 1);
 		}
 	} else {
 		VSYNC_WR_MPEG_REG_BITS(
@@ -804,6 +820,9 @@ static void vd1_set_dcu(
 	if (frame_par->hscale_skip_count)
 		r |= VDIF_CHROMA_HZ_AVG | VDIF_LUMA_HZ_AVG;
 
+	if (vf->flag & VFRAME_FLAG_VIDEO_LINEAR)
+		r |= (1 << 4);
+
 	/*enable go field reset default according to vlsi*/
 	r |= VDIF_RESET_ON_GO_FIELD;
 	VSYNC_WR_MPEG_REG(VD1_IF0_GEN_REG + vd_off, r);
@@ -1175,6 +1194,14 @@ static void vd2_set_dcu(
 		VSYNC_WR_MPEG_REG_BITS(
 			G12_VD2_IF0_GEN_REG3,
 			(burst_len & 0x3), 1, 2);
+		if (vf->flag & VFRAME_FLAG_VIDEO_LINEAR)
+			VSYNC_WR_MPEG_REG_BITS(
+				G12_VD2_IF0_GEN_REG3,
+				0, 0, 1);
+		else
+			VSYNC_WR_MPEG_REG_BITS(
+				G12_VD2_IF0_GEN_REG3,
+				1, 0, 1);
 	} else {
 		VSYNC_WR_MPEG_REG_BITS(
 			VD2_IF0_GEN_REG3 + vd_off,
@@ -1205,6 +1232,9 @@ static void vd2_set_dcu(
 	if (frame_par->hscale_skip_count)
 		r |= VDIF_CHROMA_HZ_AVG | VDIF_LUMA_HZ_AVG;
 
+	if (vf->flag & VFRAME_FLAG_VIDEO_LINEAR)
+		r |= (1 << 4);
+
 	VSYNC_WR_MPEG_REG(VD2_IF0_GEN_REG + vd_off, r);
 
 	if (type & VIDTYPE_VIU_NV21)
-- 
2.24.1

